### ISSUE TITLE
Exported symbols registered twice for mach-o binaries

### ISSUE THREAD

**Issue by op2786** [Vector35/binaryninja-api#4957]
**Version and Platform (required):**
- Binary Ninja Version: 3.6.4790-dev, 10b89271
- OS: macos
- OS Version: 14.3
- CPU Architecture: arm64

**Bug Description:**
Looks like exported symbols registered twice at the same address.

**Steps To Reproduce:**
Build the file given in the additional information section and check exports.

**Screenshots:**
<img width="532" alt="Screenshot 2024-01-24 at 10 55 51" src="https://github.com/Vector35/binaryninja-api/assets/109607896/d1fb8cd3-d2d8-4e39-a139-3432b6108b17">

Example 2 from `binaryninja` binary from version 4890-dev:
<img width="516" alt="Screenshot 2024-01-24 at 10 59 46" src="https://github.com/Vector35/binaryninja-api/assets/109607896/d27eac5a-090f-4ee7-a7e8-01a8743994d8">

**Additional Information:**
```c
#include <stdio.h>

int main() {
    printf("test\n");
    return 0;
}
```

Build with `clang++ test.cpp` on arm64 architecture.

**Comment by bpotchik** [comment]
The regression is caused by the commit `Parse Export Tries in Mach-Os`. This commit adds the entire export trie as DataSymbols, even though other symbols for the export may have already been added.

There is a separate issue and commit impacting certain ELF files as well.

**Comment by bpotchik** [comment]
Fixed in `3.6.4791-dev`

Note that this will be fixed when opening new files, however existing BNDBs will preserve the duplicate symbols.
